### PR TITLE
Scene + Transition fixes

### PR DESCRIPTION
**Pull request by fmatthew5876**
~~Depends on #1714~~

Fixes some transition issues and provides a refactor that eliminates more `Game_Temp` variables.

### CODE REVIEW

**Comment by Ghabry** [comment]
I'm not a big fan of all these
```
if
elseif
else
```
to
```
if
  return
elseif
  return
[else-code]
```
changes.

Makes it really easy to add bugs by accident when a new if-condition is added and a return is forgotten.

**Comment by fdelapena** [comment]
Diff between PR1714 and PR1717: https://github.com/fmatthew5876/Player/compare/player_scene...fmatthew5876:scene_fixes

File: src/scene.h [hunk 1]
```diff
@@ -107,14 +111,18 @@ class Scene {
 	/**
 	 * Does the transition upon starting or resuming
 	 * the scene
+	 *
+	 * @param prev_scene - the scene we transitioned from
```
Comment (Ghabry) [inline src/scene.h]: remove these dashes

Comment (fmatthew5876) [reply src/scene.h]: Fixed

**Pull request merged**